### PR TITLE
First components

### PR DESCRIPTION
**Pull request by TheVannysher**
**THIS IS IN DRAFT**

(no github premium Sadge)

this pr includes:
- [X] bottom navigator
- [X] tailwind config
- [x] vine component
- [x] vine-list component

**screenshots**

`bottom navigator`
image

### CODE REVIEW

File: web-app/src/app/components/vine/vine.component.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  @Input({ required: true }) item: Vine;
+  editing:boolean = false;
+  cepage: Observable<Sepage>;
+  tags: Observable<Tag[]>;
+
+  constructor(
+    private taskService: TasksService,
+    private cepageService: SepageService
+  ) {
+    this.tags = taskService.getTagForTask("sfkzU0IAloHlGFw5bcEG");
```
Comment (TheVannysher) [inline web-app/src/app/components/vine/vine.component.ts:25]: need to fetch the tag for the taskId (last maintenance) in the vine Object. if no task, no tags and `--` in the last maintenance field shown to the user

File: web-app/src/app/components/vine/vine.component.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            <p class="font-bold">{{item.id}}</p>
+        </div>
+        <div class="flex flex-col">
+            <div class="my-1 h-4 w-full flex justify-between">
+                <p i18n class="text-neutral-400">Grape variety: </p>
+                <p i18n class="text-neutral-400">{{(cepage | async)?.name}}</p>
+            </div>
+            <div class="my-1 h-4 w-full flex justify-between">
+                <p i18n class="text-neutral-400">last maintenance date: </p>
+                <p class="text-neutral-400">{{item.lastMaintenance}}</p>
```
Comment (TheVannysher) [inline web-app/src/app/components/vine/vine.component.html:23]: if no item.lastMaintenance, display `--` or remove the field from displayed node

File: web-app/src/app/generics/text/text.component.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<p class="default-font">
```
Comment (TheVannysher) [inline web-app/src/app/generics/text/text.component.html:1]: this component is obsolete.

File: web-app/src/app/generics/text/text.component.scss [hunk 4]
```diff
@@ -0,0 +1,4 @@
+.default-font {
```
Comment (TheVannysher) [inline web-app/src/app/generics/text/text.component.scss:1]: keep this css for global styling.

File: web-app/src/app/guards/roles/admin.guard.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { inject } from '@angular/core';
+import { CanActivateFn } from '@angular/router';
+import { AuthService } from 'src/app/services/firebase/auth/auth.service';
+
+export const adminGuard: CanActivateFn = (route, state) => {
+  const authService: AuthService = inject(AuthService);
+  authService.getLoginStatus()
```
Comment (TheVannysher) [inline web-app/src/app/guards/roles/admin.guard.ts:7]: should have a TODO here. This guard will interact with the RoleService to check if the user has the role `admin`. if not return false or redirect to another page saying this user doesn't have acces to the route.

File: web-app/src/app/guards/roles/worker.guard.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import { CanActivateFn } from '@angular/router';
+
+export const workerGuard: CanActivateFn = (route, state) => {
+  return true;
```
Comment (TheVannysher) [inline web-app/src/app/guards/roles/worker.guard.ts:4]: Same here, this guard will interact with the RoleService to determinate if the user has acces to the route or should be redirected.

File: web-app/src/app/modules/dashboard/dashboard.component.html [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+                <div class="cardContent flex justify-center items-center">
+                    <p class="mr-2">
+                        add new vine
+                    </p>
+                    <ng-icon name="featherPlus" color="black"></ng-icon>
+                </div>
+            </app-card>
+            <app-card class="max-w-[16rem]" (onClick)="openModal('createBulkVines')">
+                <div class="cardContent flex justify-center">
+                    add new vines (bulk)
```
Comment (TheVannysher) [inline web-app/src/app/modules/dashboard/dashboard.component.html:14]: will be remove. a field for number of vines added will be in the creation form for the vines.

File: web-app/src/app/modules/login/login.component.html [hunk 8]
```diff
@@ -0,0 +1,36 @@
+<div class="flex justify-center items-center w-screen h-screen">
+    <!-- for loading animation <p class="motion-reduce:hidden animate-spin">test</p> -->
```
Comment (TheVannysher) [inline web-app/src/app/modules/login/login.component.html:2]: todo: remove comment

File: web-app/.firebase/hosting.ZGlzdA.cache [hunk 9]
```diff
@@ -0,0 +1,11 @@
+favicon.ico,1692550897229,2c19690e9587bae12f419b34d2edeecc76808099540a9c9f4ea6194116cfc8f7
```
Comment (LeChuckBerry) [inline web-app/.firebase/hosting.ZGlzdA.cache:1]: `.gitignore` pls

File: web-app/.gitignore [hunk 10]
```diff
@@ -40,3 +40,10 @@ testem.log
 # System files
 .DS_Store
 Thumbs.db
+
+./src/lib/settings/dev.ts
+# Firebase
+.firebase
```
Comment (LeChuckBerry) [inline web-app/.gitignore:46]:
```suggestion
.firebase
./web-app/.firebase/*
```

File: web-app/src/app/components/tag/tag.component.html [hunk 11]
```diff
@@ -0,0 +1,10 @@
+<div
+    class="h-4 rounded-full px-3 flex justify-center gap-1 items-center relative group"
+    [ngClass]="tag.color ? 'bg-[{{tag.color}}]' : 'bg-blue-500'"
+>
+    <p *ngIf="tag.title != undefined" class="max-w-[5.5rem] w-auto truncate text-white text-xs overflow-hidden cursor-default">
+        {{tag.title}}
+    </p>
+    <ng-icon *ngIf="tag.icon != undefined" [name]="tag.icon" color="white"></ng-icon>
+    <span class="tooltip group-hover:scale-100">allo</span>
```
Comment (TheVannysher) [inline web-app/src/app/components/tag/tag.component.html:9]: put in a real tooltip pls

File: web-app/src/app/components/vine/vine.component.html (hunk 2 above)
Comment (TheVannysher) [reply web-app/src/app/components/vine/vine.component.html:26]: added to the TODO. the feature with the maintenance tied to the vine will be done in the future.

**Pull request merged**